dask/utils.py

```diff
@@ -2085,12 +2086,21 @@ def is_namedtuple_instance(obj: Any) -> bool:
 
 
 def get_default_shuffle_algorithm() -> str:
-    if d := config.get("shuffle", None):
+    if d := config.get("dataframe.shuffle.algorithm", None):
         return d
     try:
         from distributed import default_client
 
         default_client()
+        # We might lose annotations if low level fusion is active
+        if not dask.config.get("optimization.fuse.active"):
```

Comment:
I'm a bit confused about why this is needed (not necessarily saying it's wrong, just that I'm lacking context). @hendrikmakait you said 

> This failure is cause by shuffling and task fusion not playing together nicely at the moment due to our reliance on the task name being unchanged

Can you point me to where I can read more about this?

Reply:
Possibly related, are we accounting for `dask.dataframe` and `dask.array` treating this config option slightly differently? `dask.array` will perform low-level task fusion by default, while `dask.dataframe` won't. 

`dask.array` 

https://github.com/dask/dask/blob/a71c15b15e05c5896821e9c7420dfa22787a0388/dask/array/optimization.py#L52-L55

`dask.dataframe`

https://github.com/dask/dask/blob/a71c15b15e05c5896821e9c7420dfa22787a0388/dask/dataframe/optimize.py#L27-L31

Sorry for being late to the party, I'm still catching up from some recent PTO